### PR TITLE
Authnchecks

### PR DESCRIPTION
**Pull request by sgandrathi**
## Description of Change
* Added Signed install check to existing authentication check.
* Added Authorization check on admin restricted modules.
* Added a new scan parameter to take user JWT token for the tool to make requests to admin endpoints using user authentication information and check for Authorization issues.
* Updated authentication check to filter out results from post-install pages and invalid responses.
* Updated requirement variables and constants to reflect the [latest security requirements](https://developer.atlassian.com/platform/marketplace/security-requirements/#connect-apps) numbering.
* Branding analysis is set to False from the default scan parameters.
* Added a flag to scan parameters to only run Authorization check and skip all other checks.
* Added Cryptography library as a dependency for generating RSA private key.

## Fixes
* Security requirement variables and contacts are updated to reflect the [latest security requirements](https://developer.atlassian.com/platform/marketplace/security-requirements/#connect-apps) numbering.

## Did you test your changes?
- Yes, updated test cases to pass new changes.
- Added new test case for Authorization check

## Notes
- Added a call to the [MPAC](https://developer.atlassian.com/platform/marketplace/rest/v2/intro/#about) endpoint within a try/exception block.

## Reviewers
@jwong101 ,  @seanmarpo

### CODE REVIEW

File: .github/workflows/csrt-lint-and-test.yml [hunk 1]
```diff
@@ -30,9 +30,9 @@ jobs:
     - name: Lint with flake8
       run: |
         pipenv run lint
-    - name: Check dependencies for security issues
-      run: |
-        pipenv check
+#    - name: Check dependencies for security issues
+#      run: |
```
Comment (seanmarpo) [inline .github/workflows/csrt-lint-and-test.yml:34]: Highly suggest we monitor https://github.com/nabla-c0d3/sslyze for releases so we can re-enable these checks. I have email alerts setup for new releases on that project.

File: main.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
 from utils.app_validator import AppValidator
 
 
-def main(descriptor_url, skip_branding=False, debug=False, timeout=30, out_dir='out', json_logging=False):
+def main(descriptor_url, skip_branding=True, debug=False, timeout=30, out_dir='out', json_logging=False):
```
Comment (seanmarpo) [inline main.py:21]: Update the `README.md` with this new default value change

**Review by seanmarpo — CHANGES_REQUESTED** [review]
Update `README.md` with documentation updates, otherwise, looks good to me

File: main.py (hunk 2 above)
Comment (sgandrathi) [reply main.py:21]: Updated ✅ , thanks for catching it.

File: .github/workflows/csrt-lint-and-test.yml (hunk 1 above)
Comment (sgandrathi) [reply .github/workflows/csrt-lint-and-test.yml:34]: Sure, we will monitor and update it. I subscribed for the new releases on [Sslyze](https://github.com/nabla-c0d3/sslyze) project.

**Review by seanmarpo — APPROVED** [review]
👍 Looks good to me

**Review by jwong101 — APPROVED** [review]
LGTM

**Pull request merged**